cmd/reset-host/main.go

```diff
@@ -0,0 +1,102 @@
+package main
+
+import (
+	"flag"
+	"fmt"
+	"os"
+	"path/filepath"
+
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/tools/clientcmd"
+
+	metal3v1alpha1 "github.com/metal3-io/baremetal-operator/pkg/apis/metal3/v1alpha1"
+	metal3clientset "github.com/metal3-io/baremetal-operator/pkg/clientset/versioned"
+)
+
+func main() {
+	var kubeconfig *string
+	var namespace *string
+	var wipeAnnotation *bool
+
+	kubeconfig = flag.String("kubeconfig", defaultKubeConfig(), "absolute path to the kubeconfig file")
+	namespace = flag.String("n", "", "namespace for resources")
+	wipeAnnotation = flag.Bool("a", false, "wipe the status annotation")
+	flag.Parse()
+
+	if *namespace == "" {
+		fmt.Fprint(os.Stderr, "-n namespace is required\n")
+		os.Exit(1)
+	} else {
+		fmt.Printf("namespace %q\n", *namespace)
+	}
+
+	// use the current context in kubeconfig
+	config, err := clientcmd.BuildConfigFromFlags("", *kubeconfig)
+	if err != nil {
+		fmt.Fprintf(os.Stderr, "Could not build config: %s\n", err.Error())
+		os.Exit(1)
+	}
+	fmt.Printf("loaded kubeconfig from %s\n", *kubeconfig)
+
+	client, err := metal3clientset.NewForConfig(config)
+	if err != nil {
+		fmt.Fprintf(os.Stderr, "Could not build client: %s\n", err.Error())
+		os.Exit(1)
+	}
+	hostClient := client.Metal3V1alpha1().BareMetalHosts(*namespace)
+
+	for _, hostName := range flag.Args() {
+		fmt.Printf("starting %q ", hostName)
+
+		getOpts := v1.GetOptions{}
+		host, err := hostClient.Get(hostName, getOpts)
+		if err != nil {
+			fmt.Fprintf(os.Stderr, "Could not get host %q: %s\n", hostName, err)
+			continue
+		}
+
+		if *wipeAnnotation {
+			annotations := host.GetAnnotations()
+			if annotations != nil {
+				if _, ok := annotations[metal3v1alpha1.StatusAnnotation]; ok {
+					delete(annotations, metal3v1alpha1.StatusAnnotation)
+				}
+
+				fmt.Printf("annotation... ")
```

Comment:
Can this print be more descriptive? Perhaps "deleting annotation"?

Reply:
The output for each host ends up all on one line, like:

```
starting "hostname" annotation... status... done
```

when the option to clear the annotation isn't set, it is

```
starting "hostname" status... done
```